File: Source/DotNET/Model/Interactions/RaiseTriggerAction.cs
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

namespace Cratis.Scene.Model.Interactions;

/// <summary>
/// Fires a declared application trigger.
/// </summary>
/// <param name="Trigger">The application trigger to fire.</param>
public record RaiseTriggerAction(string Trigger) : InteractionAction;

File: Source/JavaScript/model/interactions/RaiseTriggerAction.ts
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

import { InteractionAction } from './InteractionAction';

/**
 * Fires a declared application trigger.
 */
export interface RaiseTriggerAction extends InteractionAction {
    trigger: string;
}

export const RaiseTriggerActionPropertyNames: (keyof RaiseTriggerAction)[] = ['trigger', 'kind'];
